### PR TITLE
Deleting network ports CP + detecting changed fields

### PR DESCRIPTION
**Pull request by clairecfu**
- [x] no more uncaught errors when remove network connections
- [x] detecting changed mac addresses
- [x] detecting adding/removed network connection changes

### CODE REVIEW

File: rackcity/utils/asset_changes_utils.py [hunk 1]
```diff
@@ -20,9 +20,26 @@ def get_changes_on_asset(asset, asset_cp):
...
+            network_port = network_ports.get(port_name=network_port_cp.port_name,)
+            if network_port.mac_address != network_port_cp.mac_address:
+                return True
+        except ObjectDoesNotExist:
+            continue
+        if network_port.mac_address != network_port_cp.mac_address:
+            return True
+    return False
+
+
```
Comment (clairecfu) [inline rackcity/utils/asset_changes_utils.py:42]: poor mac addresses we keep on forgetting them

File: rackcity/utils/asset_changes_utils.py [hunk 2]
```diff
@@ -33,22 +50,26 @@ def network_connections_have_changed(asset, asset_cp):
...
-        cp_connected_port_in_cp = network_port_cp.connected_port
-        real_connected_port_in_cp = None
-        if cp_connected_port_in_cp:
-            related_asset = cp_connected_port_in_cp.asset.related_asset
+        connected_port_cp = network_port_cp.connected_port
+        if (not connected_port_live and connected_port_cp) or (
+            connected_port_live and not connected_port_cp
+        ):
+            return True
+
```
Comment (clairecfu) [inline rackcity/utils/asset_changes_utils.py:58]: added this check to detect added/removed connections

File: rackcity/utils/asset_utils.py [hunk 3]
```diff
@@ -245,7 +245,8 @@ def handle_network_connection_delete_on_cp(port_name, asset_id, change_plan):
         destination_port_live = NetworkPort.objects.get(
             id=network_port_live.connected_port_id
         )
-        _ = copy_asset_to_new_asset_cp(destination_port_live, change_plan)
+        destination_asset_live = Asset.objects.get(id=destination_port_live.asset.id)
+        _ = copy_asset_to_new_asset_cp(destination_asset_live, change_plan)
```
Comment (clairecfu) [inline rackcity/utils/asset_utils.py:249]: so started debugging this method but then we dont actually use it anymore

File: rackcity/utils/asset_utils.py [hunk 4]
```diff
@@ -285,9 +286,7 @@ def save_network_connections(asset_data, asset_id, change_plan=None):
             failure_message += "Port name '" + port_name + "' is not valid. "
             continue
         if not network_connection_data_has_destination(network_connection):
-            # Delete network connection
-            if change_plan:
-                handle_network_connection_delete_on_cp(port_name, asset_id, change_plan)
+
```
Comment (clairecfu) [inline rackcity/utils/asset_utils.py:289]: this was causing a lot of uncaught errors-> commenting this out is actually resulting in behavior that is more expected... still not sure if we fully handle network connection deletions but think its better to just not attempt to handle it if we have uncaught errors

File: rackcity/utils/asset_changes_utils.py [hunk 5]
```diff
@@ -20,9 +20,26 @@ def get_changes_on_asset(asset, asset_cp):
         changes.append("network_connections")
     if power_connections_have_changed(asset, asset_cp):
         changes.append("power_connections")
+    if mac_addresses_have_changed(asset, asset_cp):
+        changes.append("mac_addresses")
     return changes
 
 
+def mac_addresses_have_changed(asset, asset_cp):
```
Comment (juliasaveliff) [inline rackcity/utils/asset_changes_utils.py:28]: hi mac address changes, it's so nice to finally have you here

File: rackcity/utils/asset_changes_utils.py [hunk 6]
```diff
@@ -20,9 +20,26 @@ def get_changes_on_asset(asset, asset_cp):
...
+    network_ports_cp = NetworkPortCP.objects.filter(asset=asset_cp)
+    network_ports = NetworkPort.objects.filter(asset=asset)
+    for network_port_cp in network_ports_cp:
+        try:
+            network_port = network_ports.get(port_name=network_port_cp.port_name,)
```
Comment (juliasaveliff) [inline rackcity/utils/asset_changes_utils.py:33]: weird comma but ev4 be like

File: rackcity/utils/asset_changes_utils.py [hunk 7]
```diff
@@ -20,9 +20,26 @@ def get_changes_on_asset(asset, asset_cp):
...
+            if network_port.mac_address != network_port_cp.mac_address:
+                return True
+        except ObjectDoesNotExist:
+            continue
+        if network_port.mac_address != network_port_cp.mac_address:
+            return True
```
Comment (juliasaveliff) [inline rackcity/utils/asset_changes_utils.py:39]: why r u checking twice bro

File: rackcity/utils/asset_utils.py (hunk 4 above)
Comment (juliasaveliff) [reply rackcity/utils/asset_utils.py:289]: haha. ok

File: rackcity/utils/asset_changes_utils.py (hunk 7 above)
Comment (clairecfu) [reply rackcity/utils/asset_changes_utils.py:39]: oops lol

**Pull request merged**